apps/jupyter/extensions/notebook-runner/src/task-importer.ts

```diff
@@ -36,32 +36,67 @@ const extractFolder = async (
 
   const folderPath = prefix.endsWith("/") ? prefix : `${prefix}/`;
 
-  for (const [path, file] of Object.entries(zip.files)) {
-    // Only process files within the specified folder
-    if (path.startsWith(folderPath) && !file.dir) {
-      const relativePath = path.substring(folderPath.length);
-      if (relativePath) {
-        const blob = await file.async("blob");
-        files.set(relativePath, blob);
-        console.log(`extracted from ${relativePath} from ${folderPath}`);
+  try {
+    for (const [path, file] of Object.entries(zip.files)) {
+      // Only process files within the specified folder
+      if (path.startsWith(folderPath) && !file.dir) {
+        const relativePath = path.substring(folderPath.length);
+        if (relativePath) {
+          try {
+            const blob = await file.async("blob");
+            files.set(relativePath, blob);
+          } catch (error) {
+            throw new FileSystemError(
+              "extract",
+              path,
+              error instanceof Error ? error : undefined,
+            );
+          }
+        }
       }
     }
+  } catch (error) {
+    if (error instanceof FileSystemError) {
+      throw error;
+    }
+
+    throw new FileSystemError(
+      "read folder",
```

Comment:
Do you want to be more specific here? Or alternative do we want to use an enum instead of a string?

Reply:
Forgotten @bubudetp?